ibtk/include/ibtk/FEDataManager.h

```diff
@@ -618,6 +618,53 @@ class FEDataManager : public SAMRAI::tbox::Serializable
     /*!
      * \brief Spread a density from the FE mesh to the Cartesian grid using the
      * default spreading spec.
+     *
+     * @note This function may spread forces from points near the physical
+     * boundary into ghost cells outside the physical domain. To account for
+     * these forces one should usually call
+     * IBTK::RobinPhysBdryPatchStrategy::accumulateFromPhysicalBoundaryData()
+     * after this function to move said force values into the physical domain.
+     */
+    void spread(int f_data_idx,
+                libMesh::NumericVector<double>& F,
+                libMesh::NumericVector<double>& X,
+                const std::string& system_name);
+
+    /*!
+     * \brief Spread a density from the FE mesh to the Cartesian grid.
+     *
+     * @param[in] f_data_idx SAMRAI index into the
+     * SAMRAI::hier::PatchHierarchy object owned by this class. The spread
+     * force values will be added into this index.
+     * @param[in] F Finite element solution vector containing the field which
+     * will be spread onto the Eulerian grid.
+     * @param[in] X Finite element solution vector containing the current
+     * position of the mesh.
+     * @param[in] system_name name of the system corresponding to @p F.
+     *
+     * Both @p X and @p F should contain ghost values corresponding to the IB
+     * partitioning of the Lagrangian data, i.e., vectors returned from
+     * buildIBGhostedVector.
+     *
+     * @note This function may spread forces from points near the physical
+     * boundary into ghost cells outside the physical domain. To account for
+     * these forces one should usually call
+     * IBTK::RobinPhysBdryPatchStrategy::accumulateFromPhysicalBoundaryData()
+     * after this function to move said force values into the physical domain.
+     */
+    void spread(int f_data_idx,
+                libMesh::NumericVector<double>& F,
+                libMesh::NumericVector<double>& X,
+                const std::string& system_name,
+                const SpreadSpec& spread_spec);
+
+    /*!
+     * \brief Spread a density from the FE mesh to the Cartesian grid using the
+     * default spreading spec.
+     *
+     * @deprecated This version of spread is deprecated - users should close
```

Comment:
I would prefer to keep some "simple" spreading routines that do not require manual use of physical boundary operator objects. What is the rationale for deprecating?

Reply:
Is there some helper code to make it easy to apply those operators at patch boundaries across a level and/or hierarchy?

Reply:
I deprecated them since they are the wrong choice when one wants to use multiple parts - they require using a scratch index and using the boundary operator for each part.

Reply:
Are there helper functions to help to do this correctly with the scratch hierarchy?

Reply:
There are not.

Lets fix this by just not deprecating the convenience routines.